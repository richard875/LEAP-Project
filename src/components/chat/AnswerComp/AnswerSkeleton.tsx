import { useContext, useRef, useEffect } from "react";

import { Skeleton } from "@/components/ui/skeleton";
import ConversationContext from "@/context/conversationContext";

import AnswerPill from "./AnswerPill";

const AnswerSkeleton = () => {
  const bottomRef = useRef<HTMLDivElement>(null);
  const context = useContext(ConversationContext);

  useEffect(() => {
    if (context?.loading && bottomRef.current) {
      bottomRef.current.scrollIntoView({
        behavior: "smooth",
        block: "end",
        inline: "end",
      });
    }
  }, [context?.loading]);

  return (
    <div ref={bottomRef} className="mt-5 mb-5">
      <AnswerPill />
      <div className="space-y-2">
        <Skeleton className="h-4 max-w-[300px]" />
        <Skeleton className="h-4 max-w-[450px]" />
        <Skeleton className="h-4 w-full" />
        <Skeleton className="h-4 w-full" />
      </div>
    </div>
  );
};

export default AnswerSkeleton;
