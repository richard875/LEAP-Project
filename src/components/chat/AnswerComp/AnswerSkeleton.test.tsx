import { render } from "@testing-library/react";

import "@testing-library/jest-dom";
import AnswerSkeleton from "./AnswerSkeleton";

describe("AnswerSkeleton", () => {
  it("should render correctly", () => {
    const { container } = render(<AnswerSkeleton />);
    expect(container).toBeTruthy();
  });

  it("should match the snapshot", () => {
    const { asFragment } = render(<AnswerSkeleton />);
    expect(asFragment()).toMatchSnapshot();
  });
});
