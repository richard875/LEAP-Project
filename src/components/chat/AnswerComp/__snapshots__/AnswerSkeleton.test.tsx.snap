// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnswerSkeleton should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="mt-5 mb-5"
  >
    <div
      class="w-fit mb-5 h-fit pl-2.5 pr-2.5 pt-2 pb-2 rounded-full border border-stone-300 dark:border-stone-500 flex items-center justify-center gap-2.5 select-none"
    >
      <img
        alt="LawConnect Logo"
        class="select-none"
        data-nimg="1"
        decoding="async"
        height="20"
        src="/_next/image?url=%2Flogo-small-dark.png&w=48&q=75"
        srcset="/_next/image?url=%2Flogo-small-dark.png&w=32&q=75 1x, /_next/image?url=%2Flogo-small-dark.png&w=48&q=75 2x"
        style="color: transparent;"
        width="20"
      />
      <p
        class="text-sm font-medium"
      >
        LawConnect
      </p>
    </div>
    <div
      class="space-y-2"
    >
      <div
        class="bg-accent animate-pulse rounded-md h-4 max-w-[300px]"
        data-slot="skeleton"
      />
      <div
        class="bg-accent animate-pulse rounded-md h-4 max-w-[450px]"
        data-slot="skeleton"
      />
      <div
        class="bg-accent animate-pulse rounded-md h-4 w-full"
        data-slot="skeleton"
      />
      <div
        class="bg-accent animate-pulse rounded-md h-4 w-full"
        data-slot="skeleton"
      />
    </div>
  </div>
</DocumentFragment>
`;
